lib/path/tests/unit.nix

```diff
@@ -3,9 +3,42 @@
 { libpath }:
 let
   lib = import libpath;
-  inherit (lib.path) subpath;
+  inherit (lib.path) append subpath;
 
   cases = lib.runTests {
+    testAppendExample1 = {
+      expr = append /foo "bar/baz";
+      expected = /foo/bar/baz;
+    };
+    testAppendExample2 = {
+      expr = append /foo "./bar//baz/./";
+      expected = /foo/bar/baz;
+    };
+    testAppendExample3 = {
+      expr = append /. "foo/bar";
+      expected = /foo/bar;
+    };
+    testAppendExample4 = {
+      expr = (builtins.tryEval (append "/foo" "bar")).success;
+      expected = false;
+    };
+    testAppendExample5 = {
+      expr = (builtins.tryEval (append /foo /bar)).success;
```

Comment:
Maybe reference the design document or add a short explanation?
Same for `"/bar"` below.

Reply:
The documentation for `append` already has short explanations for all the examples, and the tests here just verify the output of those examples. I'll add a comment on the top to point out that these "Example[0-9]+" test cases correspond to the examples in the documentation.